packages/webapp/app/routes/_auth.setup.tsx

```diff
@@ -1,93 +1,51 @@
-import { Link, useNavigate, useSearchParams } from "@remix-run/react";
-import { apiBaseUrl } from "../lib/url";
-import {
-    DB_PASSWORD_KEY,
-    DB_USERNAME_KEY,
-    checkDBCredentials,
-    setupDBSync,
-} from "../lib/db";
-import { useEffect, useState } from "react";
+import { Link } from "@remix-run/react";
+import { ActionArgs, json, redirect } from "@remix-run/node";
+import { z } from "zod";
+import { createUserAccount } from "../lib/auth.server";
+import { commitSession, getSession } from "../sessions.server";
 
-export default function AuthSignup() {
-    const [searchParams] = useSearchParams();
-    const navigate = useNavigate();
+const zForm = z.object({
+    username: z.string().nonempty(),
+    password: z.string().nonempty(),
+});
 
-    const tokenInit = searchParams.get("token");
+export async function action({ request }: ActionArgs) {
```

Comment:
This should fail, and possibly redirect if there's already a user account created.

Reply:
Limitation added and created #18 to better inform users.